frontend/app/src/pages/BuySell/AddLiquidity.tsx

```diff
@@ -48,10 +48,10 @@ const AddLiquidityComponent: React.FC<WithTranslation> = ({ t }) => {
 
   const calcMaxToken = (slippage: number, cashDeposited: number) => {
     if (cfmmStorage) {
-      const { tokenPool, cashPool } = cfmmStorage;
+      const { tezPool, cashPool } = cfmmStorage;
       const cash = cashDeposited * 1e6;
       const max =
-        Math.ceil(((cash * tokenPool.toNumber()) / cashPool.toNumber()) * (1 + slippage * 0.01)) /
+        Math.ceil(((cash * cashPool.toNumber()) / tezPool.toNumber()) * (1 + slippage * 0.01)) /
```

Comment:
This makes the wrong assumption about price

Reply:
you need to use the functions that compute what the actual price obtained is, using Newton descent etc

Reply:
Ah nm this is for adding liquidity